Mathlib/Topology/Algebra/Group/Basic.lean

```diff
@@ -1522,6 +1522,33 @@ theorem TopologicalGroup.t2Space_of_one_sep (H : ∀ x : G, x ≠ 1 → ∃ U 
 #align topological_group.t2_space_of_one_sep TopologicalGroup.t2Space_of_one_sep
 #align topological_add_group.t2_space_of_zero_sep TopologicalAddGroup.t2Space_of_zero_sep
 
+/-- Given a neighborhood `U` of the identity, one may find a neighborhood `V` of the identity which
+is closed, symmetric, and satisfies `V * V ⊆ U`. -/
+@[to_additive]
+theorem exists_nhds_one_isClosed_inv_eq_mul_subset {U : Set G} (hU : U ∈ 𝓝 1) :
+    ∃ V ∈ 𝓝 1, IsClosed V ∧ V⁻¹ = V ∧ V * V ⊆ U := by
+  have A : (fun p : G × G => p.1 * p.2) ⁻¹' U ∈ 𝓝 ((1, 1) : G × G) :=
+    continuousAt_fst.mul continuousAt_snd (by simpa)
+  simp only [nhds_prod_eq, Filter.mem_prod_self_iff, prod_subset_iff, mem_preimage] at A
+  rcases A with ⟨V, V_mem, hV⟩
+  rcases exists_mem_nhds_isClosed_subset V_mem with ⟨W, W_mem, W_closed, hW⟩
+  refine ⟨W ∩ W⁻¹, Filter.inter_mem W_mem (inv_mem_nhds_one G W_mem), W_closed.inter W_closed.inv,
+    by simp [inter_comm], ?_⟩
+  calc
+  W ∩ W⁻¹ * (W ∩ W⁻¹)
```

Comment:
I could do that, but it wouldn't help here since the point is to get a neighborhood which is both symmetric and closed, and for this I also need a regular space. I could have a lemma like that in a space with `[ContinuousInv G] [RegularSpace G]`, but that feels quite articifical.

Reply:
Oh right you would need to know that the symmetrization preserves closedness for my idea to work. It does of course but that's definitely not worth adding.